devito/passes/iet/dtypes.py

```diff
@@ -2,8 +2,44 @@
 import ctypes
 
 from devito.ir import FindSymbols, Uxreplace
+from devito.ir.iet.nodes import Dereference
+from devito.tools.utils import as_tuple
+from devito.types.basic import Symbol
 
-__all__ = ['lower_complex']
+__all__ = ['lower_scalar_half', 'lower_complex']
+
+
+def lower_scalar_half(iet, lang, sregistry):
```

Comment:
Ok, will refactor. I suppose `lower_dtypes` will first handle the float16 scalars, then call `_complex_headers` for the includes, then finally apply the primary `types` mapping?

Reply:
not sure why three steps, just one step that does the type mapping then on step that does the headers/...

Reply:
True, I can group the type mappings but I do need to do the scalars in their own loop before the other mappings (to avoid them getting mapped along with any array/indexed/whatever else, since those mapped types don't work for scalars)